### PR TITLE
This pull request is to merge the game_logic branch to Development branch 

### PR DESCRIPTION
**Pull request by henatan99**
In this milestone the game_logic is completely defined with the in the TicTacToe game class. player and board objects are also constituents of the game class along with various method definitions.

### CODE REVIEW

File: bin/main [hunk 1]
```diff
@@ -1,99 +1,125 @@
...
+  end
+
+  def win_move?
+    @win_triplets = [[1, 2, 3], [4, 5, 6], [7, 8, 9], [1, 4, 7], [2, 5, 8], [3, 6, 9], [1, 5, 9], [3, 5, 7]]
+    @win_triplets.each do |triplet|
+      return true if [@board.symbol[triplet[0] - 1], @board.symbol[triplet[1] - 1],
+                      @board.symbol[triplet[2] - 1]].all?(@current_player.sym) == true
+    end
+  end
+end
```
Comment (SpaYco) [inline bin/main]: - [ ] All game-logic related resources (classes and/or modules) are kept in the `lib` directory. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone1/lib_dir_exist.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

File: lib/player.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class Player
+  attr_accessor :name, :game, :sym
```
Comment (SpaYco) [inline lib/player.rb]: - [ ]  `attr_accessor` isn't needed here, and you should be fine with `attr_reader`/`attr_writer`, please change it.

**Review by SpaYco — CHANGES_REQUESTED** [review]
# Changes Requested 

Hello Team! 👋 
Awesome job so far!! 👍 you still need to make some changes though! 

## Required Changes ♻️ 

- [ ] the game should print that the game is a draw when no one wins

- [ ] more below 👇 

great work on making the README file 👏 

### please make the appropriate changes and ask for another code review ✔️ 

Happy Coding!

Best, 
Aziz Mejri 
_contact me on [twitter](https://twitter.com/iSpaYco) or slack (@SpaYco) if you need any help_

**Review by krys2fa — CHANGES_REQUESTED** [review]
## Excellent work @henatan99 and @zdrale 👏 
Well done implementing the changes requested by the reviewer. 💪 
Kudos for having a README file set up even though it isn't required yet as well. 👍 

Just a few minor edge cases to consider to complete the milestone. They can be seen below:
### Status: Changes Required ♻️ 
- [ ] [REQUIRED] No `puts/print` and `gets.chomp` statements are allowed outside `main.rb` file. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/only_main_print.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/)). You currently have files in your lib folder with `puts` statements. Let's fix this.


- [ ] [REQUIRED] The game ends (as it should) but does not explicitly state that it has ended in a draw even when all the cells have been filled and there is no winner. See an example in the screenshot below.
Screen Shot 2020-11-26 at 6 48 32 PM


- [ ] [REQUIRED] The game ends unexpectedly in a draw even when there is still a chance of a player winning the game. See the screenshot below:
Screen Shot 2020-11-26 at 6 42 51 PM

That's it! Please request for another review after implementing the changes.

Happy Coding 🖥️  😃 

###### You can contact me on slack (Chris Amanor) for any clarification.

**Review by voscarmv — APPROVED** [review]
# Project status: Approved ✨ 😎👍 💯 ✔️

- Excellent work correctly applying all of the changes requested by the previous TSE.
- I see you have a typo in your *'continuue'* message (line 137 in `main`). Please make sure to fix it before merging!

### You may merge this branch and move forward ⛙🛣
### Questions or comments? Reach me on [Twitter](https://twitter.com/voscarmv) or [LinkedIn](https://linkedin.com/in/oscar-mier) 🧑‍💻🗨️🧑‍💻 
# Happy Hacking! 🧠 ⌨️ 🖱️ 🖥️

**Pull request merged**